arch/xtensa/src/esp32/esp32_spi.c

```diff
@@ -46,18 +46,33 @@
 #include "esp32_spi.h"
 #include "esp32_gpio.h"
 #include "esp32_cpuint.h"
+#include "esp32_dma.h"
 
 #include "xtensa.h"
 #include "hardware/esp32_gpio_sigmap.h"
 #include "hardware/esp32_dport.h"
 #include "hardware/esp32_spi.h"
 #include "hardware/esp32_soc.h"
+#include "hardware/esp32_pinmap.h"
 #include "rom/esp32_gpio.h"
 
 /****************************************************************************
  * Private Types
  ****************************************************************************/
 
+/* SPI DMA hardware channel number */
+
+#define SPI_DMA_CHAN      (2)
+
+/* SPI DMA RX/TX description number */
+
+#define SPI_DMADESC_NUM   (CONFIG_SPI_DMADESC_NUM)
```

Comment:
```suggestion
#define SPI_DMADESC_NUM   (CONFIG_SPI_DMADESC_NUM + 1)
```
Shouldn't be +1 for the last descriptor?

Reply:
The driver add EOF and "next=NULL" into to the last DMA description to tell hardware that this is the last node, so that we need to add en extra one.